### DISCUSSION TITLE
YARP IP address problem

### DISCUSSION THREAD

**Discussion by SSSSamZhu** [robotology/community#582] [Q&A] [answered]
Hi ，
After I deleted the installed yarp related files, YARP seems to be installed in a correct way, and `yarpmanager` can be started normally. However, after I connected iCub to the network cable, YARP seemed not to start normally. I have the impression that I should modify the configuration file, could you tell me where I should modify it ?
```
Setting up yarp and iCub env vars
icub@isaacliang-LC3:~$ yarpserver 

     ==========================
    | __  __ ___  ____   ____  |
    | \ \/ //   ||  _ \ |  _ \ |
    |  \  // /| || |/ / | |/ / |
    |  / // ___ ||  _ \ |  _/  |
    | /_//_/  |_||_| \_\|_|    |
    |                          |
     ==========================

Call with --help for information on available options
Using port database: :memory:
Using subscription database: :memory:
IP address: default
Port number: 10000
[WARNING] |yarp.os.MultiNamespace| Could not find namespace the_robot_network_namespace
[ERROR] |yarp.os.Port|/| Port / failed to activate at tcp://192.168.1.101:10000/ (address conflict)
[ERROR] Name server failed to open


```
Thanks！

PS:In addition, it is possible to run `yarpserver` for a short time after restarting the computer and entering the graphical interface. I tried running `yarp conf --clean` to reset the IP address, but it didn't work

**Comment by traversaro** [comment]
Are you sure that there is not a yarpserver already running somewhere?

**Reply by SSSSamZhu** [reply]
I thought it was the same at first, and I even ended up rebooting my computer.The problem occurred after I connected to the icub robot's network cable.Because I want to connect to icub. Actually, I think my problem is more that the green arrow on the `yarpmanager` doesn't light up.Clicking on him didn't do anything .
![Screenshot from 2022-06-30 19-42-39](https://user-images.githubusercontent.com/100264795/176669147-78e4597e-f711-4547-aa72-fe26d943546b.png)

**Comment by pattacini** [comment, marked as answer]
Normally, you don't have to run a `yarpserver` locally to your PC if you connect it to a pre-existing robot network, which has its own YARP server.

What you need to do is, instead, to align the namespace of your local PC to that namespace of the robot.

Usually, the namespace of the robot is `/icub`. To find it out, just ssh into `icub-head` and type in `yarp namespace`.
Once you have confirmed it, change the namespace of your PC by doing (replace `/icub` below with whatever name you have found):
```console
>> yarp namespace /icub
```
Finally, you could test your connection by means of:
```console
>> yarp where
```

**Reply by SSSSamZhu** [reply]
Hi @pattacini 
I did the operation as shown in the figure, but I didn't seem to expect the result.
![Screenshot from 2022-07-04 10-32-38](https://user-images.githubusercontent.com/100264795/177071517-8e24c695-5bc9-4ab7-bbd5-600ce60b36e4.png)

**Reply by pattacini** [reply]
Well, the output of `yarp where` seems to be:
```console
Name server /root is available at IP 192.168.1.101 port 10000
```
So it's fine 👍🏻
You could ask for the list of ports. 
```console
>> yarp name list
```

Don't launch a second instance of `yarpserver` on your machine then, when it's connected to the robot network.

**Reply by SSSSamZhu** [reply]
> You could ask for the list of ports.
> 
>  `yarp name list`

I've run this command before， to get registration information of all known ports. But I am not quite sure why the print result seems to be different from the ideal. 
https://www.yarp.it/latest/group__yarp.html#name_list

![Screenshot from 2022-07-04 14-34-12](https://user-images.githubusercontent.com/100264795/177095544-2316ba95-25cd-4273-aa30-23e27e906841.png)

**Reply by pattacini** [reply]
Probably, the robot is off and there are no YARP ports open yet, although it's strange that `yarp name list` won't be printing the server's port at least.

Try to launch the robot then and repeat the procedure.

I can still see that you're trying to launch `yarpserver` from your PC when connected to the robot network. It is not required and it's fundamentally wrong as there's already a server. Indeed, as you can notice, there's an `address conflict` generated because you're attempting to launch a `yarpserver` with the same namespace of another instance already running in the network. So, don't do that.

**Reply by SSSSamZhu** [reply]
Thanks, I think I see what you mean. Maybe I need to reopen the robot's YARP port?It should also be for this reason that iCub cannot be connected using yarpmanager.
![Screenshot from 2022-07-04 15-15-45](https://user-images.githubusercontent.com/100264795/177101914-dbadd99a-7e76-456f-b281-24575becb373.png)

**Reply by pattacini** [reply]
From your log, I can clearly notice that `icub-head` (aka the robot) is unable to find the `yarpserver`. This is getting kind of puzzling @SSSSamZhu.

Thus, let's start over.
Can you tell us what the layout of your robot network looks like?
How many machines the robot is connected to via Ethernet?

Also, to make things simpler and if that's not already done, you may make sure that each machine of the robot network can ssh into any other with passwordless login.
To this end, you'd need to rely on the `ssh-copy-id` command. Here are some references:
- https://linuxhint.com/use-ssh-copy-id-command/
- https://icub-tech-iit.github.io/documentation/icub_operating_systems/pc104/the-icub-live-installation/?h=ssh+copy+id#ssh-passwordless-login

**Reply by SSSSamZhu** [reply]
Hi @pattacini 

> Thus, let's start over.
> Can you tell us what the layout of your robot network looks like?
> How many machines the robot is connected to via Ethernet?

We plan to connect two devices on the iCub robot, one is a Linux host (the network cable port used by the laptop is the laptop matching the robot before), and the other is a Windows host. However, DURING my debugging, I unplug the Windows network cable to prevent network port conflicts. 


> you may make sure that each machine of the robot network can ssh into any other with passwordless login.
`ssh-copy-id` I did fellowing in linux.
I don't seem to find the guidance document when OPERATING on window, and I can't open yarpServer and YarpManager either.https://github.com/robotology/robotology-superbuild/issues/1189

```
Setting up yarp and iCub env vars
icub@isaacliang-LC3:~$ ssh-keygen -t rsa
Generating public/private rsa key pair.
Enter file in which to save the key (/home/icub/.ssh/id_rsa): 
/home/icub/.ssh/id_rsa already exists.
Overwrite (y/n)? Y
Enter passphrase (empty for no passphrase): 
Enter same passphrase again: 
Your identification has been saved in /home/icub/.ssh/id_rsa
Your public key has been saved in /home/icub/.ssh/id_rsa.pub
The key fingerprint is:
SHA256:sdVZYnG2a67cc+rNt/NhwNVAjKqNZt6J3Y7nTuqyw78 icub@isaacliang-LC3
The key's randomart image is:
+---[RSA 3072]----+
|            +=*  |
|           o.*.o.|
|        . ..o ...|
|         +. . .. |
|        S+   oo  |
|        = .  o.  |
|       = + o. .o |
|        * +=ooo++|
|        .*E**.+=O|
+----[SHA256]-----+

icub@isaacliang-LC3:~$ ssh-copy-id -i /home/icub/.ssh/id_rsa.pub icub@icub-head
/usr/bin/ssh-copy-id: INFO: Source of key(s) to be installed: "/home/icub/.ssh/id_rsa.pub"
/usr/bin/ssh-copy-id: INFO: attempting to log in with the new key(s), to filter out any that are already installed
/usr/bin/ssh-copy-id: INFO: 1 key(s) remain to be installed -- if you are prompted now it is to install the new keys

Number of key(s) added: 1

Now try logging into the machine, with:   "ssh 'icub@icub-head'"
and check to make sure that only the key(s) you wanted were added.

icub@isaacliang-LC3:~$ ssh-copy-id -i /home/icub/.ssh/id_rsa.pub icub@isaacliang-LC3
/usr/bin/ssh-copy-id: INFO: Source of key(s) to be installed: "/home/icub/.ssh/id_rsa.pub"
/usr/bin/ssh-copy-id: INFO: attempting to log in with the new key(s), to filter out any that are already installed
/usr/bin/ssh-copy-id: INFO: 1 key(s) remain to be installed -- if you are prompted now it is to install the new keys

Number of key(s) added: 1

Now try logging into the machine, with:   "ssh 'icub@isaacliang-LC3'"
and check to make sure that only the key(s) you wanted were added.

icub@isaacliang-LC3:~$ ssh icub@icub-head
Welcome to iCub OS version 8.0.18.04-20190124 based on Ubuntu 18.04.2 LTS (GNU/Linux 4.15.0-96-lowlatency x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

  System information as of Thu Apr 16 03:30:42 UTC 2020

  System load:  0.02               Processes:                172
  Usage of /:   39.3% of 26.51GB   Users logged in:          0
  Memory usage: 1%                 IP address for enp0s31f6: 10.0.1.104
  Swap usage:   0%                 IP address for main-bond: 192.168.1.111


 * Canonical Livepatch is available for installation.
   - Reduce system reboots and improve kernel security. Activate at:
     https://ubuntu.com/livepatch

131 packages can be updated.
0 updates are security updates.

Failed to connect to https://changelogs.ubuntu.com/meta-release-lts. Check your Internet connection or proxy settings


Last login: Thu Apr 16 03:25:57 2020 from 192.168.1.101
Setting up yarp and iCub env vars
```

**Reply by SSSSamZhu** [reply]
Hi @pattacini
When checking the installation I noticed that there was an error reported here and I am not sure if this is the cause of the problem. I'm trying to configure the system from scratch.
Thank you for your patience. Looking forward to your reply.

 
```
[ 98%] Completed 'icub-basic-demos'
[100%] Built target icub-basic-demos
icub@isaacliang-LC3:~/Desktop/robot/robotology-superbuild/build$ yarpdev --list
[INFO] |yarp.dev.Drivers| Here are devices listed for your system:
[WARNING] |yarp.dev.Drivers| Wrong library name for plugin readonlyremotecontrolboard
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[ERROR] |yarp.os.YarpPluginSettings| Error while opening /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so:
  /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_fakeFrameGrabber.so: undefined symbol: test_grabber
[WARNING] |yarp.dev.Drivers| Wrong library name for plugin test_grabber
[INFO] |yarp.dev.Drivers| Device "AudioPlayerWrapper", available on request (found in /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_AudioPlayerWrapper.so library), is a network wrapper.
[INFO] |yarp.dev.Drivers| Device "AudioRecorderWrapper", available on request (found in /home/icub/Desktop/robot/robotology-superbuild/build/install/lib/yarp/yarp_AudioRecorderWrapper.so library), is a network wrapper.
...

```

**Reply by pattacini** [reply]
Let's take the Windows machine out of the equation for the moment.

I didn't actually understand if you aim to replace our [iCub Server Console][console] with a brand new Linux machine or just wanted to add up one machine more.

My suggestion is to keep our server laptop (usually called `icubsrv`) in place and plug your Linux host as a further machine in the network.

1. Make sure that the alias of the new host is fine and recognizable from anywhere. Quick test: could you ping `icub-head`, `icubsrv`, `your-host` from each machine? If not, get the name of your host (use `hostname`) and [edit `/etc/hosts`][hosts] everywhere.
2. Make sure that yarp namespace is `/root` on all the 3 machines: `icub-head` + `icubsrv` + your host.
3. Start from a clean configuration (power cycle all the machines).
4. Launch `yarpserver` on `icubsrv`.
5. Connect to `icub-head` and launch `yarp where` to check whether you're able to see the YARP server running on `icubsrv`.
6. Connect to your host and do the same. If you're unsuccessful, you could try to fix the problem with
    - `yarp conf <icubsrv-ip> 10000`.
    - Once done, try again with `yarp where`. You should succeed now.
7. At this point you may start up the robot by launching `yarprobotinterface` on the `icub-head`.
8. If the robot is up and running, do `yarp name list` on your host to see the list of YARP ports opened up by the robot itself.
9. Shut down the robot by hitting CTRL+C on the same console where you ran `yarprobotinterface`.
10. On `icubsrv` locate the `cluster-config.xml` file by doing `yarp resource --context yarpmanager --from cluster-config.xml`.
11. Edit that file to make sure that it contains the 3 machines.
12. Now, you should be able to launch `yarpmanager` correctly. 

[console]: https://icub-tech-iit.github.io/documentation/icub_operating_systems/other-machines/icub-server-laptop/
[hosts]: https://www.baeldung.com/linux/mapping-hostnames-ports#:~:text=In%20Linux%2C%20%2Fetc%2Fhosts,use%20for%20testing%20websites%20locally.

**Reply by SSSSamZhu** [reply]
Hi @pattacini ,

Before I start replying, please allow me to explain our basic IP information：
192.168.1.111 `icub-head  pc104 `
192.168.1.100 `icubsrv`
192.168.1.101 `isaacliang-LC3(our host)   `

> I didn't actually understand if you aim to replace our [iCub Server Console](https://icub-tech-iit.github.io/documentation/icub_operating_systems/other-machines/icub-server-laptop/) with a brand new Linux machine or just wanted to add up one machine more.
> My suggestion is to keep our server laptop (usually called `icubsrv`) in place and plug your Linux host as a further machine in the network.

In our initial plan, we planned to use  `isaacliang-LC3 (our host)` instead of `icubsrv`. But this may not be a good solution, we will install your suggested implementation. Thank you very much for your advice.

> 
> 1. Make sure that the alias of the new host is fine and recognizable from anywhere. Quick test: could you ping `icub-head`, `icubsrv`, `your-host` from each machine? If not, get the name of your host (use `hostname`) and [edit `/etc/hosts`](https://www.baeldung.com/linux/mapping-hostnames-ports#:~:text=In%20Linux%2C%20%2Fetc%2Fhosts,use%20for%20testing%20websites%20locally.) everywhere.
> 2. Make sure that yarp namespace is `/root` on all the 3 machines: `icub-head` + `icubsrv` + your host.
> 3. Start from a clean configuration (power cycle all the machines).
> 4. Launch `yarpserver` on `icubsrv`.
> 5. Connect to `icub-head` and launch `yarp where` to check whether you're able to see the YARP server running on `icubsrv`.

Up to this point everything was going well, as expected. Two of each machine can ping each other. And as before, `icubsrv` and `isaacliang-LC3` can also be SSH into `icub-head`.

> 6. Connect to your host and do the same. If you're unsuccessful, you could try to fix the problem with
>    * `yarp conf <icubsrv-ip> 10000`.
>    * Once done, try again with `yarp where`. You should succeed now.

I tried to run 'yarpserver' on `isaacliang-LC3` and then execute `yarp where` on `icub-head`, this doesn't seem to be successful, `icub-head` seems to be listening on `icubsrv` all the time, While ignoring `isaacliang-LC3`. It is also possible that I misunderstood you and performed the wrong operation. But surprisingly, this seems to solve our original problem.

> 7. At this point you may start up the robot by launching `yarprobotinterface` on the `icub-head`.
> 8. If the robot is up and running, do `yarp name list` on your host to see the list of YARP ports opened up by the robot itself.

I tried to run `yarpserver` on `icubsrv` and 'Isaacliang-LC3' at the same time, I know this is not a wise move, but I found that the only thing that could communicate with `icub-head` was`icubsrv`, After I turn `icubsrv` `yarpserver` off, `icub-head` does not execute `yarprobotInterface`. Unfortunately, I didn't have a screenshot at that time, I can supplement it if necessary later.

> 9. Shut down the robot by hitting CTRL+C on the same console where you ran `yarprobotinterface`.
> 10. On `icubsrv` locate the `cluster-config.xml` file by doing `yarp resource --context yarpmanager --from cluster-config.xml`.
> 11. Edit that file to make sure that it contains the 3 machines.

I don't know how to modify this file, this is the screenshot after I execute the command.
![Screenshot from 2022-07-08 17-05-36](https://user-images.githubusercontent.com/100264795/177964217-822a35b0-0c93-43c3-8c28-2d0d0a742430.png)

![Screenshot from 2022-07-08 17-07-51](https://user-images.githubusercontent.com/100264795/177964175-3d25efdf-d39e-41ec-815c-6e5e9b7b9d16.png)

> 12. Now, you should be able to launch `yarpmanager` correctly.

Thanks, it's works, but there are still a few hiccups. It seems that the` icub-head` connection issue has not been resolved.
![Screenshot from 2022-07-08 17-12-05](https://user-images.githubusercontent.com/100264795/177964112-952c674f-a302-49f2-9e19-c8b2d362e232.png)
Thank you for your patience. Looking forward to your reply.

**Reply by pattacini** [reply]
>I tried to run 'yarpserver' on isaacliang-LC3 and then execute yarp where on icub-head, this doesn't seem to be successful, icub-head seems to be listening on icubsrv all the time, While ignoring isaacliang-LC3. 

Well, this might be because the YARP configuration on `icub-head` still points to `icubsrv`. Thus, on `icub-head` you could do:
```console
>> yarp conf --clean
>> yarp conf IP 10000
```
where `IP` is the IP address of `isaacliang-LC3`. Find out more on [`yarp conf`](https://www.yarp.it/latest/group__yarp.html#yarp_conf).

Remember, only one instance of `yarpserver` can run on the network.

Once you have launched the `yarpserver` on `isaacliang-LC3` and done with the `yarp conf` things above, you should be able finally to find the server from the `icub-head` with a `yarp where`.

>I don't know how to modify this file, this is the screenshot after I execute the command.

You ought to be doing this on the machine from which you launch `yarpmanager`, which seems to be `isaacliang-LC3` and NOT `icubsrv` from your last screenshot.

Also, the latter screenshot shows that you have `pc104` instead of `icub-head` within the file `cluster-config.xml`. So:
- find the file via `yarp resource...`;
- edit the file to replace `pc104` with `icub-head`.

**Reply by SSSSamZhu** [reply]
Hi, @pattacini 
Thank you very much.My issue has been solved.
Also, would you like to give me a brief introduction to how I should configure Windows host? I might need it to communicate with a Linux host. My original intention was to use this module.
https://github.com/ami-iit/yarp-device-openxrheadset

**Reply by pattacini** [reply]
Hi @SSSSamZhu 

Happy it did work!
For a Windows machine, just install the software[^1] and then configure the machine via the `yarp conf` commands reported above.

[^1]: My suggestion is that you install the SW via the [robotology-superbuild](https://icub-tech-iit.github.io/documentation/sw_installation/windows_from_sources_superbuild/).

**Reply by SSSSamZhu** [reply]
Hi, @pattacini
I'm sorry to bother you again
> Happy it did work! For a Windows machine, just install the software[1](#user-content-fn-1-64811411d033ae94abb22f8150dcba5f) and then configure the machine via the `yarp conf` commands reported above.
> 1. My suggestion is that you install the SW via the [robotology-superbuild](https://icub-tech-iit.github.io/documentation/sw_installation/windows_from_sources_superbuild/). [↩](#user-content-fnref-1-64811411d033ae94abb22f8150dcba5f)

I wonder if I need to complete this SSH step? I saw that the documentation seemed to be only for Linux, and there was no guide to the Console Server for Windows. https://icub-tech-iit.github.io/documentation/icub_operating_systems/other-machines/icub-server-laptop/?h=namesp#install-the-ssh-keys-for-password-less-login-on-icub-head
I have installed the [robotology-superbuild](https://icub-tech-iit.github.io/documentation/sw_installation/windows_from_sources_superbuild/) [↩](https://github.com/robotology/community/discussions/582#user-content-fnref-1-64811411d033ae94abb22f8150dcba5f)yet.  I can also ping to other hosts now, but the data cannot be sent out.

**Reply by pattacini** [reply]
When [OpenSSH](https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse) became available on Windows we tried it out and discovered that not all the features were usable: for example, the passwordless connection wasn't working at that time. We haven't followed up on this recently, but, generally speaking, we don't support SSH connectivity on Windows machines inside our ecosystem yet.

For this and other reasons, we don't even support Console Server solutions for Windows.

A Windows machine can be still connected to a YARPed cluster quite easily though.

First off, configure the YARP network using the `yarp conf` command as described in the posts above. There's no difference wrt a Linux OS.

Second, if you want to launch remotely a YARP module on a Windows machine straight from a `yarpmanager` instance running on a Linux host, do:
- Make sure that the `yarpserver` is running ok on a Linux machine.
- Make sure that you can see the server from the Windows machine (use `yarp conf...` and `yarp where` commands).
- Run `yarprun --server /windows` on the Windows machine. You can replace `/windows` with whatever tag you like.
- You can now refer to `/windows` as a node inside your applications.

Hope this helps
 
cc @Nicogene @traversaro

**Reply by SSSSamZhu** [reply]
Hi, @pattacini
> Indeed, as you can notice, there's an `address conflict` generated because you're attempting to launch a `yarpserver` with the same namespace of another instance already running in the network. So, don't do that.

I'm beginning to see what you mean. As an example from my current situation, my iCub robot is connected to a total of three computers, which are laptop + workstation +Windows. I named them icubsrv, isaacLiang-LC3 and Windows respectively. If I run `yarp conf 192.168.1.101 10000` (IP address of isaacLiang-LC)in the icub-head configuration, will I no longer be able to run `yarpserver` on either Windows or icubsrv(of course, isaacLiang-LC can be used continuously)? This will cause address conflicts, other hosts need to be added to the cluster as nodes if they are to be used. 

Thank you very much.

**Reply by pattacini** [reply]
Hi @SSSSamZhu 

Only one instance of the `yarpserver` is required to be running on a YARP network. According to your notes above, just run `yarpserver` on isaacLiang-LC and configure the remaining machines to check the server therein using `yarp conf`. I'm not really getting the point...
